### DISCUSSION TITLE
Sourcing code blocks from an internal file

### DISCUSSION THREAD

**Discussion by mhuzzell** [IBAHCM/RPiR#555] [Q&A for project] [answered]
I would like to just make a single block of code to source from demo to demo, to generate the tree populations. I know how to do this in an R 'project', but the file structure of the packages is _very_ formal, and I'm worried about disrupting that. Should such a file just go in the demo folder, and get called as "demo/filename"? Or in the general whole-package folder, and just called as "filename"? Or somewhere else? 

Thank you!

**Comment by richardreeve** [comment, marked as answer]
@mhuzzell - it's an interesting idea to do that. The simplest solution is to put it into the `demo` folder as you suggest. Alternatives include [putting it into an `inst` folder](https://r-pkgs.org/misc.html#sec-misc-inst) - the contents of that folder get automatically installed with the packages, but can be hard to source those files correctly so don't worry if you find that too complex. The final(?) alternative is to write a function that returns the tree populations, which is handled just like all of the other functions you write in a package.
